ui/src/components/SessionProvider.tsx

```diff
@@ -27,8 +27,32 @@ export default function SessionProvider({
   const [session, setSession, clearSession] = useLocalStorage('session', null);
 
   useEffect(() => {
+    const clearSessionIfNecessary = async () => {
+      const config = await getConfig();
+      if (session && session.required === false) {
+        let authEnabled = false;
+        if (
+          config.authentication.methods.oidc.enabled ||
+          config.authentication.methods.github.enabled ||
+          config.authentication.methods.jwt.enabled
+        ) {
+          authEnabled = true;
+        }
+        if (config.authentication.required === true || authEnabled) {
+          clearSession();
+        }
+      }
+    };
+
     const loadSession = async () => {
-      let session = {
+      if (session) {
+        clearSessionIfNecessary();
+        if (session) {
```

Comment:
is this block required? either way it would return right?

Reply:
@markphelps I didn't want the rest of the code in that function to run if there exists a session after we've tried to clear it

Reply:
ah yeah i didnt see that teh function continued